ts-force/src/rest/restHeader.ts

```diff
@@ -0,0 +1,30 @@
+export type SforceMru = {
+  updateMru: boolean;
+};
+export type PackageVersion = {
+  package: string;
+  version: string;
+};
+
+export type GenericHeader = {
+  header: string;
+  value: string;
+};
+
+export type RequestHeadersInput = {
+  'Sforce-Auto-Assign'?: boolean;
+  'Sforce-Mru'?: SforceMru;
+  'x-sfdc-packageversion'?: PackageVersion;
+  'If-Modified-Since'?: Date;
+  'If-Unmodified-Since'?: Date;
+  'If-Match'?: string[];
+  'If-None-Match'?: string[];
+  'Generic-Header'?: GenericHeader;
+};
+
+export const ConditionalRequestHeaders: (keyof RequestHeadersInput)[] = [
+  'If-Match',
+  'If-None-Match',
```

Comment:
I've never actually used these `etag` caching headers with Salesforce... Reviewing [the docs](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_sobject_retrieve_get.htm), I see this statement:

> An optional header specifying a comma-delimited list of one or more ETags. This only has an effect when used with Account objects. The request is only processed if the Account’s ETag does not match one of the ETags in the list.

Does it really only work with `Account`?

Reply:
From the document [here](https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_get_started_intro.htm), Etag is used for http caching. I guess that’s the reason why it’s said only work with Account. I test with a a case project, and `If-Match` and `If-None-Match` can be added on get request but not on patch request.
<img width="600" alt="Screenshot 2024-02-23 at 9 20 14 AM" src="https://github.com/vacasaoss/ts-force/assets/74746314/c36d169a-90ff-4454-8b65-3f1b3f474cf2">
<img width="600" alt="Screenshot 2024-02-23 at 9 19 46 AM" src="https://github.com/vacasaoss/ts-force/assets/74746314/8414d32b-6f39-4f9b-a81f-90e602389a5b">